weldx/asdf/file.py

```diff
@@ -429,18 +473,16 @@ def write_to(
         The given input file name or a buffer, in case the input was omitted.
 
         """
-        created = False
         if fd is None:
             fd = BytesIO()
-            created = True
 
         # if no args are given, we use the specifications given in the constructor.
         if not write_args:
             write_args = self._write_kwargs
 
         self._asdf_handle.write_to(fd, **write_args)
 
-        if created:
+        if isinstance(fd, types_file_like.__args__):
```

Comment:
interesting!
(I suppose the `.__args__` is needed here?)

Reply:
yes, it is kind of ugly, but works as expected. I mean what is the use of all of this type annotation stuff, if we could not use it for runtime type checking in the first place :)